subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,224 @@
+	crlCmdRevokeDeviceCa  = "revoke-device-ca"
+	crlCmdDisableDeviceCa = "disable-device-ca"
+)
+
+func init() {
+	revokeCmd := &cobra.Command{
+		Use:   "revoke-device-ca <PKI Directory>",
+		Short: "Revoke device CA from the list of CAs allowed to issue device client certificates",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Revoke device CA from the list of CAs allowed to issue device client certificates.
+
+When the online or local device CA is revoked:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
+		Annotations: map[string]string{crlCmdAnnotation: crlCmdRevokeDeviceCa},
+	}
+	caCmd.AddCommand(revokeCmd)
+	addRevokeCmdFlags(revokeCmd, "revoke")
+
+	disableCmd := &cobra.Command{
+		Use:   "disable-device-ca <PKI Directory>",
+		Short: "Disable device CA devices with client certificates issued by it can no longer be registered",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Disable device CA devices with client certificates issued by it can no longer be registered.
+
+When the online or local device CA is disabled:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can continue to connect and use FoundriesFactory.
+
+Usually, when the device CA is compromised, a user would:
+1. Immediately disable a given device CA using "fioctl keys ca disable-device-ca <PKI Directory> --serial <CA Serial>".
+2. Inspect their devices with client certificates issued by that device CA, and remove any compromised devices.
+3. Create a new device CA using "fioctl keys ca add-device-ca <PKI Directory> --online-ca|--local-ca".
+4. Rotate a client certificate of all legitimate devices from step 2 to the certificate issued by a new device CA.
+5. Revoke a given device CA using "fioctl keys ca revoke-device-ca <PKI Directory> --serial <CA Serial>".`,
+		Annotations: map[string]string{crlCmdAnnotation: crlCmdDisableDeviceCa},
+	}
+	caCmd.AddCommand(disableCmd)
+	addRevokeCmdFlags(disableCmd, "disable")
+}
+
+func addRevokeCmdFlags(cmd *cobra.Command, op string) {
+	cmd.Flags().BoolP("dry-run", "", false,
+		"Do not "+op+" the certificate, but instead show a generated CRL that will be uploaded to the server.")
+	cmd.Flags().BoolP("pretty", "", false,
+		"Can be used with dry-run to show the generated CRL in a pretty format.")
+	cmd.Flags().StringArrayP("ca-file", "", nil,
+		"A file name of the device CA to "+op+". Can be used several times to "+op+" several device CAs")
```

Comment:
```suggestion
		"A file name of the device CA to "+op+". Can be used multiple times to "+op+" multiple device CAs")
```

Reply:
Not sure about that suggestion. "Multiple" sounds more natural to me, but after some googling, looks like both terms would be OK, just a matter of style

Reply:
I find `several device CAs` more appropriate here than `multiple device CAs`, pointing at a normally small (limited) number of device CAs per factory.
However, it makes sense to use `_multiple_ times to revoke _several_ device CAs` to avoid tautology.

Reply:
https://github.com/foundriesio/fioctl/pull/324/commits/ee087eaf9f830d0f5475335f818c472e24338e47